src/unread/unreadPmsReducers.js

```diff
@@ -19,10 +19,21 @@ export default (state: UnreadState = initialState, action: Action): UnreadState
     case ACCOUNT_SWITCH:
       return initialState;
 
-    case EVENT_NEW_MESSAGE:
-      return action.message.type === 'private' && action.message.display_recipient.length === 2
-        ? addItemsToPmArray(state, [action.message.id], action.message.sender_id)
-        : state;
+    case EVENT_NEW_MESSAGE: {
+      if (action.message.type !== 'private') {
+        return state;
+      }
+
+      if (action.ownEmail && action.ownEmail === action.message.sender_email) {
+        return state;
+      }
```

Comment:
Nice to see you are on top of commits even in `zulip-mobile`. :)

Hmm, there isn't a `flags.read` structure though. Maybe this is not included on new message server event? I don't think there is any case where the new message will be already read besides one sent by ourselves?

I'll do the change everywhere, replacing the check for ownEmail with ownUserId.